### PR TITLE
[FIX] CodeQL reports

### PR DESCRIPTION
**Pull request by eseiler**
These are all `Comparison of narrow type with wide type in loop condition`

### CODE REVIEW

**Comment by seqan-actions** [comment]
Documentation preview available at https://docs.seqan.de/preview/seqan/seqan3/3307

File: test/unit/utility/math_test.cpp [hunk 1]
```diff
@@ -36,8 +36,8 @@ TYPED_TEST(unsigned_operations, floor_log2)
     for (uint8_t log2_value = 0; log2_value < seqan3::detail::bits_of<unsigned_t>; ++log2_value)
     {
         unsigned_t start = unsigned_t{1u} << log2_value;
-        unsigned_t end = start << 1u;
-        for (unsigned_t n = start, k = 0u; n < end && k < max_iterations; ++n, ++k)
+        unsigned_t end = static_cast<unsigned_t>(std::min<size_t>(start / 2u, max_iterations));
+        for (unsigned_t n = start; n < end; ++n)
```
Comment (SGSSGene) [inline test/unit/utility/math_test.cpp:40]: There seems to be something wrong here.
end = start * 2
but also the result of std::min seems wrong:
I expeected something like:
`std::min(start, max_iterations) + start`

File: test/unit/utility/math_test.cpp [hunk 2]
```diff
@@ -65,13 +65,13 @@ TYPED_TEST(unsigned_operations, ceil_log2)
     for (uint8_t log2_value = 0; log2_value < seqan3::detail::bits_of<unsigned_t>; ++log2_value)
     {
         unsigned_t start = unsigned_t{1u} << log2_value;
-        unsigned_t end = start << 1u;
+        unsigned_t end = static_cast<unsigned_t>(std::min<size_t>(start / 2u, max_iterations));
```
Comment (SGSSGene) [inline test/unit/utility/math_test.cpp:68]: same here

**Review by SGSSGene — CHANGES_REQUESTED** [review]

File: test/unit/utility/math_test.cpp (hunk 1 above)
Comment (eseiler) [reply test/unit/utility/math_test.cpp:40]: Yes, that was very wrong 😆

**Comment by eseiler** [comment]
BTW, reports are here https://github.com/seqan/seqan3/security/code-scanning

**Review by SGSSGene — APPROVED** [review]

**Pull request merged**